MezfliprSup/Mezflipr_single.db

```diff
@@ -14,141 +15,160 @@ record(ai, "$(P)$(NAME):COMPENSATION") {
-    field(OUT, "@Mezflipr.proto setConst($(CODE)) $(PORT)")
+record(mbbi, "$(P)$(NAME):MODE") {
+    field(DESC, "$(NAME) Mode")
+    field(INP, "@Mezflipr.proto getMode $(PORT)")
+    field(SCAN, "1 second")
     field(DTYP, "stream")
-    field(PREC, "2")
+    info(interest, "HIGH")
+    info(archive, "VAL")
+    
+    field(ZRST, "static")
+    field(ONST, "steps")
+    field(TWST, "analytical")
+    field(THST, "file")
     
     field(SIML, "$(P)SIM")
-    field(SIOL, "$(P)$(NAME):SIM:CONSTANT:SP")
+    field(SIOL, "$(P)$(NAME):SIM:MODE")
     field(SDIS, "$(P)DISABLE")
 }
 
-record(ai, "$(P)$(NAME):DT") {
-    field(DESC, "$(NAME) Delta T")
-    field(INP, "@Mezflipr.proto getDt($(CODE)) $(PORT)")
-    field(SCAN, "1 second")
+record(ao, "$(P)$(NAME):CURRENT:SP") {
+    field(DESC, "$(NAME) Current SP")
+    field(OUT, "@Mezflipr.proto setCurrent $(PORT)")
     field(DTYP, "stream")
     field(PREC, "2")
-    field(EGU, "us")
-    info(interest, "HIGH")
-    info(archive, "VAL")
+    field(EGU, "A")
     
     field(SIML, "$(P)SIM")
-    field(SIOL, "$(P)$(NAME):SIM:DT")
+    field(SIOL, "$(P)$(NAME):SIM:CURRENT:SP")
     field(SDIS, "$(P)DISABLE")
-    
-    field(HIHI, "0.0000000001")
-    field(HHSV, "MAJOR")
 }
 
-record(ao, "$(P)$(NAME):DT:SP") {
-    field(DESC, "$(NAME) Delta T SP")
-    field(OUT, "@Mezflipr.proto setDt($(CODE)) $(PORT)")
-    field(DTYP, "stream")
-    field(PREC, "2")
-    field(EGU, "us")
+record(waveform, "$(P)$(NAME):CURRENT_STEPS:SP") {
+    field(DESC, "$(NAME) Current steps SP")
+    
+    field(FTVL, "CHAR")
+    field(NELM, "1024")
     
     field(SIML, "$(P)SIM")
-    field(SIOL, "$(P)$(NAME):SIM:DT:SP")
+    field(SIOL, "$(P)$(NAME):SIM:CURRENT_STEPS:SP")
     field(SDIS, "$(P)DISABLE")
     
-    field(DRVH, "0")
-    field(DRVL, "-99999999999")
+    field(FLNK, "$(P)$(NAME):CURRENT_STEPS:SP:_SEND")
 }
 
-record(stringin, "$(P)$(NAME):FILENAME") {
-    field(DESC, "$(NAME) Filename")
-    field(INP, "@Mezflipr.proto getFile($(CODE)) $(PORT)")
-    field(SCAN, "1 second")
+record(bo, "$(P)$(NAME):CURRENT_STEPS:SP:_SEND") {
+    field(OUT, "@Mezflipr.proto setCurrentSteps($(P)$(NAME)) $(PORT)")
     field(DTYP, "stream")
-    info(interest, "HIGH")
-    info(archive, "VAL")
+}
+
+record(waveform, "$(P)$(NAME):CURRENT_ANALYTICAL:SP") {
+    field(DESC, "$(NAME) Current steps SP")
+    
+    field(FTVL, "CHAR")
+    field(NELM, "1024")
     
     field(SIML, "$(P)SIM")
-    field(SIOL, "$(P)$(NAME):SIM:FILENAME")
+    field(SIOL, "$(P)$(NAME):SIM:CURRENT_ANALYTICAL:SP")
     field(SDIS, "$(P)DISABLE")
+    field(FLNK, "$(P)$(NAME):CURRENT_ANALYTICAL:SP:_SEND")
 }
 
-record(stringout, "$(P)$(NAME):FILENAME:SP") {
-    field(DESC, "$(NAME) Filename SP")
-    field(OUT, "@Mezflipr.proto setFile($(CODE)) $(PORT)")
+record(bo, "$(P)$(NAME):CURRENT_ANALYTICAL:SP:_SEND") {
+    field(OUT, "@Mezflipr.proto setCurrentAnalytical($(P)$(NAME)) $(PORT)")
     field(DTYP, "stream")
+}
```

Comment:
As discussed, try doing this as one record or put a note on stream device gotchas on wiki.

Reply:
@FreddieAkeroyd and I tried this again with both `lso` and writing waveforms directly to streamdevice but could not get either one to work. I've documented how to do this at https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Stream-Device-Tips-and-Tricks

Reply:
Could you also post on tech-talk as it looks like this should work from documentation?